When (re)starting a node that's already configured with some Sources, it could happen that some transient warning logs could be logged because metrics were missing if said Sources started to receive traffic too soon.  This has been fixed.

Example of such logs:

```
2025-01-08T07:48:36.421822+00:00 [warning] tag: RESOURCE, msg: handle_resource_metrics_failed, reason: {badkey,received}, stacktrace: [{erlang,map_get,[received,#{}],[{error_info,#{module => erl_erts_errors}}]},{emqx_metrics_worker,idx_metric,4,[{file,"emqx_metrics_worker.erl"},{line,562}]},{emqx_metrics_worker,inc,4,[{file,"emqx_metrics_worker.erl"},{line,262}]},{emqx_resource_metrics,handle_telemetry_event,4,[{file,"emqx_resource_metrics.erl"},{line,127}]},{telemetry,'-execute/3-fun-0-',4,[{file,"telemetry.erl"},{line,167}]},{lists,foreach_1,2,[{file,"lists.erl"},{line,1686}]},{emqx_bridge_mqtt_connector,on_message_received,3,[{file,"emqx_bridge_mqtt_connector.erl"},{line,70}]},{emqx_bridge_mqtt_ingress,handle_match,5,[{file,"emqx_bridge_mqtt_ingress.erl"},{line,279}]},{lists,foreach_1,2,[{file,"lists.erl"},{line,1686}]},{emqx_bridge_mqtt_ingress,handle_publish,3,[{file,"emqx_bridge_mqtt_ingress.erl"},{line,260}]},{emqtt,eval_msg_handler,3,[{file,"emqtt.erl"},{line,2038}]},{emqtt,deliver,3,[{file,"emqtt.erl"},{line,2015}]},{emqtt,connected,3,[{file,"emqtt.erl"},{line,1332}]},{gen_statem,loop_state_callback,11,[{file,"gen_statem.erl"},{line,1395}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,241}]}], event: received, kind: error, hint: transient failures may occur when restarting a resource, resource_id: <<"source:mqtt:tset:connector:mqtt:test">>
```
